### PR TITLE
Update release notes for v3.16.4

### PR DESCRIPTION
**Pull request by rene-dekker**

### CODE REVIEW

File: calico-enterprise_versioned_docs/version-3.16/_includes/release-notes/_v3.16.4-release-notes.mdx [hunk 1]
```diff
@@ -1,10 +1,17 @@
 25 September 2023
 
+### Improvements
+
+- Added federation metrics to Typha, similar to our current federation metrics in kube-controllers.
+
 ### Bug fixes
 - Fixed a security issue.
+- Fixed an issue where the use of subPath in a volumeMount could prevent pod creation on Windows when Docker is the runtime.
+- Added a decoupling buffer between the Typha client library and downstream code. This prevents the Typha client from being blocked if the downstream code blocks for too long, and prevents timeouts at very high scale.
```
Comment (bmckercher123) [inline calico-enterprise_versioned_docs/version-3.16/_includes/release-notes/_v3.16.4-release-notes.mdx:10]:
```suggestion
- Added a decoupling buffer between the Typha client library and downstream code. This prevents the Typha client from being blocked if the downstream code blocks for too long, and prevents large-scale timeouts.
```

**Review by bmckercher123 — APPROVED** [review]
Suggestion only. LGTM

File: calico-enterprise_versioned_docs/version-3.16/_includes/release-notes/_v3.16.4-release-notes.mdx (hunk 1 above)
Comment (rene-dekker) [reply calico-enterprise_versioned_docs/version-3.16/_includes/release-notes/_v3.16.4-release-notes.mdx:10]: This RN was taken verbatim from v3.15.4. I think that your suggestion changes the meaning of the sentence, am I correct?

Comment (bmckercher123) [reply calico-enterprise_versioned_docs/version-3.16/_includes/release-notes/_v3.16.4-release-notes.mdx:10]: Not sure what the original means.

**Pull request merged**